src/identity/quartzOrganizations/reducers/index.ts

```diff
@@ -19,27 +21,46 @@ export default (state = initialState, action: Actions): QuartzOrganizations =>
     switch (action.type) {
       case SET_QUARTZ_ORGANIZATIONS: {
         draftState.orgs = action.quartzOrganizations
+        draftState.status = RemoteDataState.Done
+
         return
       }
+
       case SET_QUARTZ_ORGANIZATIONS_STATUS: {
         draftState.status = action.status
         return
       }
 
       case SET_QUARTZ_DEFAULT_ORG: {
-        const {oldDefaultOrgId, newDefaultOrgId} = action
-
-        if (oldDefaultOrgId !== newDefaultOrgId) {
-          draftState.orgs.forEach(org => {
-            if (org.id === oldDefaultOrgId) {
-              org.isDefault = false
-            }
-
-            if (org.id === newDefaultOrgId) {
-              org.isDefault = true
-            }
-          })
+        // No existing default org is acceptable; oldDefaultOrg may be undefined.
+        const oldDefaultOrg = draftState.orgs.find(
```

Comment:
A totally valid criticism of this approach is that we could probably ditch both the `find()` and the `forEach()` loop if we just normalized the array of orgs we get from the API into an object with the `orgIds` as strings, and stored the current default `orgId` as a property so that we could do direct lookup on specific orgs. I could see that being a useful way to have state organized for some purposes, but for now I wouldn't implement the reducer that way for these reasons:

1. The `find()` and `forEach()` aren't expensive in a meaningful way for the user. They're not happening in a render, and the user only clicks 'change default' once in a while. This is getting run in response to a specific click, not page load.
2. Orgs are expected to be in an array format in the UI and in the clockface components, so we'd need to de-normalize the object anyway, which probably erases the performance benefit.
3. I think it'd be overengineering this reducer. This isn't doing that much work, so it doesn't need 2x as much code.